scripts/asm-installer/asm_vm

```diff
@@ -1110,9 +1115,36 @@ this project, run:
   --project="${PROJECT_ID}"
 EOF
     fi
+
+    info "Verifying the GCE image..."
+    local SOURCE_IMAGE_PATH="$(echo "$(retry 2 run gcloud compute instance-templates \
+      describe "${SOURCE_INSTANCE_TEMPLATE}" \
+    --format=json \
+    --project="${PROJECT_ID}")" \
+    | jq '.properties.disks[0] | .initializeParams.sourceImage')"
+
+    if [[ "${SOURCE_IMAGE_PATH}" == *projects/*-cloud/global/images* ]]; then
+    # check whether the public image is supported
+      check_supported_vm_linux_distro "${SOURCE_IMAGE_PATH}"
```

Comment:
You can simply have the function return true or false instead of having it set a variable.

Reply:
See `is_cluster_registered` function in this file for reference.